strongbox-data-service/src/main/java/org/carlspring/strongbox/data/domain/GenericEntity.java

```diff
@@ -101,5 +99,9 @@ public int hashCode()
         }
         return objectId.hashCode();
     }
-    
+
+    public void copyVersionTo(final GenericEntity target) {
+        target.entityVersion = this.entityVersion;
```

Comment:
@fuss86 just interesting, why this needed? :)

Reply:
To protect `configuration` from concurrent modification. Because of SB-930 we will be storing `String` representation of  `configuration` in the database (`BinaryConfiguration`), but we will be operating in the system on the real `Configuration` instance (once fetched from the database and deserialized). I need to keep  `BinaryConfiguration` instance versioned in the database, so e.g. once configuration was saved with `@OVersion` 2, we should not allow to save it obsolete version hold somewhere concurrently in the memory with `@OVersion` 1.

Reply:
And now it's changed to `copyTrackingFields` (see https://github.com/strongbox/strongbox/pull/605/files#diff-629f6d989f9f5c214414afe24219d1e7R103)